src/stackedtime/sparse.jl

```diff
@@ -0,0 +1,199 @@
+##################################################################################
+# This file is part of StateSpaceEcon.jl
+# BSD 3-Clause License
+# Copyright (c) 2020-2023, Bank of Canada
+# All rights reserved.
+##################################################################################
+
+
+# selected sparse linear algebra library is a Symbol
+const sf_libs = (
+    :default,   #
+    :umfpack,   # use Julia's standard library (UMFPACK)
+    :pardiso,   # use Pardiso - the one included with MKL
+)
+
+global sf_default = :umfpack
+
+"""
+    use_umfpack()
+
+Set the default sparse factorization library to UMFPACK (the one used in Julia's
+standard library). See also [`use_pardiso`](@ref).
+
+"""
+@inline use_umfpack() = (global sf_default = :umfpack; nothing)
+# @inline use_umfpack(m::Model) = use_umfpack!(m)
+
+"""
+    use_umfpack!(model)
+
+Instruct the stacked-time solver to use Pardiso with this model. See also
+[`use_pardiso!`](@ref).
+"""
+@inline use_umfpack!(m::Model) = (m.options.factorization = :umfpack; m)
```

Comment:
Will this macro print the model after running? Wouldn't we want it to print the factorization or nothing?

Reply:
I'm never completely sure about what to return from ! functions in Julia. My strategy is to have them return the thing being mutated, unless multiple things are being mutated, then return nothing.  Here I'm going for syntax like 
`m = use_umfpack!(m)`.

Reply:
I guess we're being a little inconsistent.
`update_links!(m)` returns `m.parameters`.
`clear_sstate!(m)` returns nothing.
`sssolve!(m)` returns the steadystate values.
`solve!(m)` returns `m`.
`linearize!(m)` returns `m`.
`setoption!(m, :shift, 10)` returns `m.options`.

I think in this case it would be most consistent to return m.options or m.options.factorization. Returning nothing would also be a good option. I think the point of the return in these functions is simply to print something to the REPL which helps visually confirm the new state of the object.

> Here I'm going for syntax like `m = use_umfpack!(m).`

I think that is confusing; an object is either manipulated or reassigned. the above syntax implies that one could write `m2 = use_umfpack!(m)` without changing `m`. That said, maybe this is a common convention?

Reply:
On second thought, one would need it to return `m` if one wanted a syntax like `sssolve!(use_umfpack!(m))`. Not sure if that is what we want, but its a thought.

Reply:
> On second thought, one would need it to return `m` if one wanted a syntax like `sssolve!(use_umfpack!(m))`. Not sure if that is what we want, but its a thought.

I think this is exactly the idea behind returning the instance being mutate - so that you can daisy chain mutating calls. 

> I guess we're being a little inconsistent. `update_links!(m)` returns `m.parameters`. `clear_sstate!(m)` returns nothing. `sssolve!(m)` returns the steadystate values. `solve!(m)` returns `m`. `linearize!(m)` returns `m`. `setoption!(m, :shift, 10)` returns `m.options`.

You're right - we are being inconsistent.  And you're also right that the return values of these functions were motivated by the provision of relevant feedback in the REPL during interactive use. 

I think we should stick with providing visual feedback - I doubt anyone would ever want to daisy chain `use_umfpack`, and being user-friendly is paramount for us.